sdk/python/tests/dsl/component_tests.py

```diff
@@ -12,32 +12,328 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-
+import kfp
 from kfp.dsl._component import component
 from kfp.dsl._metadata import ComponentMeta, ParameterMeta, TypeMeta
-from kfp.dsl._types import GCSPath, Integer
+from kfp.dsl._types import GCSPath, Integer, InconsistentTypeException
+from kfp.dsl import ContainerOp, Pipeline
 import unittest
-import mock
 
 class TestPythonComponent(unittest.TestCase):
 
-  def test_component(self):
-    """Test component decorator."""
+  def test_component_metadata(self):
+    """Test component decorator metadata."""
 
     class MockContainerOp:
       def _set_metadata(self, component_meta):
         self._metadata = component_meta
 
     @component
-    def componentA(a: {'Schema': {'file_type': 'csv'}}, b: {'number': {'step': 'large'}} = 12, c: GCSPath(path_type='file', file_type='tsv') = 'gs://hello/world') -> {'model': Integer()}:
+    def componentA(a: {'Schema': {'file_type': 'csv'}}, b: Integer() = 12, c: GCSPath(path_type='file', file_type='tsv') = 'gs://hello/world') -> {'model': Integer()}:
       return MockContainerOp()
 
-    containerOp = componentA(1,2,3)
+    containerOp = componentA(1,2,c=3)
 
     golden_meta = ComponentMeta(name='componentA', description='')
     golden_meta.inputs.append(ParameterMeta(name='a', description='', param_type=TypeMeta(name='Schema', properties={'file_type': 'csv'})))
-    golden_meta.inputs.append(ParameterMeta(name='b', description='', param_type=TypeMeta(name='number', properties={'step': 'large'}), default=12))
+    golden_meta.inputs.append(ParameterMeta(name='b', description='', param_type=TypeMeta(name='Integer'), default=12))
     golden_meta.inputs.append(ParameterMeta(name='c', description='', param_type=TypeMeta(name='GCSPath', properties={'path_type':'file', 'file_type': 'tsv'}), default='gs://hello/world'))
     golden_meta.outputs.append(ParameterMeta(name='model', description='', param_type=TypeMeta(name='Integer')))
 
     self.assertEqual(containerOp._metadata, golden_meta)
+
+  def test_type_check_with_same_representation(self):
```

Comment:
Do you have test that covers pipeline level param with metadata against component params type check?

Reply:
Great point. In fact, by adding this test, I found a bug and I fixed it.